extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java

```diff
@@ -114,6 +93,42 @@ public boolean canRender(Widget w) {
         return null;
     }
 
+    private String buildRow(Selection w, String lab, String cmd, Item item, State state) throws RenderException {
+        String rowSnippet = getSnippet("selection_row");
+
+        String command = cmd != null ? cmd : "";
+        String label = lab;
+
+        if (item instanceof NumberItem && ((NumberItem) item).getDimension() != null) {
+            String unit = getUnitForWidget(w);
+            command = StringUtils.replace(command, UnitUtils.UNIT_PLACEHOLDER, unit);
+            label = StringUtils.replace(label, UnitUtils.UNIT_PLACEHOLDER, unit);
+
+            // Special treatment for °C since uom library uses a single character: ℃
+            // This will ensure the current state matches the cmd and the buttonClass is set accordingly.
+            command = StringUtils.replace(command, "°C", "℃");
+        }
+
+        rowSnippet = StringUtils.replace(rowSnippet, "%item%", w.getItem() != null ? w.getItem() : "");
+        rowSnippet = StringUtils.replace(rowSnippet, "%cmd%", StringEscapeUtils.escapeHtml(command));
+        rowSnippet = StringUtils.replace(rowSnippet, "%label%",
+                label != null ? StringEscapeUtils.escapeHtml(label) : "");
+
+        State compareMappingState = state;
+        if (state instanceof QuantityType) { // convert the item state to the command value for proper
+                                             // comparison and "checked" attribute calculation
+            compareMappingState = convertStateToLabelUnit((QuantityType<?>) state, command);
+        }
+
+        if (compareMappingState.toString().equals(command)) {
+            rowSnippet = StringUtils.replace(rowSnippet, "%checked%", "checked=\"true\"");
+            mappedValue = (label != null) ? label : command;
```

Comment:
Why do we assign the `command` here if the `label` is `null`? We don't check for that in Basic UI.
See https://github.com/eclipse/smarthome/pull/5667/files#diff-955d3a6399e421b68151ddb7351de8bbR144

Reply:
A label for a StateOption can be null, not its value.
For the mappings in the sitemap, the syntax forces you to define a label and a value.
So to be clear, this test was useless before my change but is now required.
So it is now ok here but I have to change Basic UI.